src/box/box.cc

```diff
@@ -4296,3 +4297,264 @@ box_read_ffi_enable(void)
 	if (--box_read_ffi_disable_count == 0)
 		box_read_ffi_is_disabled = false;
 }
+
+/**
+ * Encode a sequence of values according to format string into a buffer
+ * allocated on region.
+ */
+static const char *
+mp_format_on_region(struct region *region, const char *format, ...)
+{
+	va_list ap;
+	va_start(ap, format);
+	size_t buf_size = mp_vformat(NULL, 0, format, ap);
+	char *buf = (char *)region_alloc(region, buf_size);
+	va_end(ap);
+	if (buf == NULL) {
+		diag_set(OutOfMemory, buf_size, "region_alloc", "buf");
+		return buf;
+	}
+	va_start(ap, format);
+	mp_vformat(buf, buf_size, format, ap);
+	va_end(ap);
+	return buf;
+}
+
+/**
+ * Drop the constraint or foreign key. If fieldno is equal to UINT32_MAX then
+ * the tuple constraint or tuple foreign key is dropped, otherwise the field
+ * constraint or field foreign key is dropped.
+ */
+static int
+constraint_drop(uint32_t space_id, const char *name, uint32_t name_len,
+		uint32_t fieldno, enum tuple_constraint_type type)
+{
+	const struct space *space = space_by_id(space_id);
+	uint32_t total_count;
+	struct tuple_constraint_def *cdefs;
+	if (fieldno != UINT32_MAX) {
+		assert(fieldno < space->def->field_count);
+		total_count = space->def->fields[fieldno].constraint_count;
+		cdefs = space->def->fields[fieldno].constraint_def;
+	} else {
+		total_count = space->def->opts.constraint_count;
+		cdefs = space->def->opts.constraint_def;
+	}
+	bool is_exists = false;
+	uint32_t count = 0;
+	for (uint32_t i = 0; i < total_count; ++i) {
+		if (cdefs[i].type != type)
+			continue;
+		++count;
+		uint32_t len = MIN(cdefs[i].name_len, name_len);
+		if (memcmp(cdefs[i].name, name, len) == 0)
+			is_exists = true;
+	}
+	if (!is_exists) {
+		int errcode = type == CONSTR_FKEY ? ER_NO_SUCH_FOREIGN_KEY :
+			      ER_NO_SUCH_CONSTRAINT;
+		diag_set(ClientError, errcode, tt_cstr(name, name_len),
+			 space_name(space));
+		return -1;
+	}
+
+	char key[16];
+	char *key_end = key + mp_format(key, 16, "[%u]", space_id);
+	struct region *region = &fiber()->gc;
+	uint32_t used = region_used(region);
+	const char *type_str = tuple_constraint_type_strs[type];
+	const char *path = fieldno != UINT32_MAX ?
+			   tt_sprintf("format[%d].%s", fieldno + 1, type_str) :
```

Comment:
`%d` -> `%u`. Same in `constraint_create()`.

Reply:
Fixed.